cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/error/mapper/ClouderaManagerStorageErrorMapper.java

```diff
@@ -0,0 +1,120 @@
+                        "permissions must, at a minimum, allow to assume Data Access Role and Ranger Audit Role." +
+                        "Refer to Cloudera documentation at %s for the required rights.",
+                auditLocation, dataAccessRole, rangerAuditRole, auditLocation, instanceProfile, instanceProfile,
+                "https://docs.cloudera.com/management-console/cloud/environments/topics/mc-idbroker-minimum-setup.html");
+    }
+
+    private String azureError(CloudStorage cloudStorage) {
+
+        String managedIdentity = "";
+
+        for (CloudIdentity cloudIdentity : cloudStorage.getCloudIdentities()) {
+            if (CloudIdentityType.ID_BROKER == cloudIdentity.getIdentityType()) {
+                managedIdentity = cloudIdentity.getAdlsGen2Identity().getManagedIdentity();
+            }
+        }
+
+        String auditLocation = getRangerAuditDir(cloudStorage);
+        AccountMapping accountMapping = cloudStorage.getAccountMapping();
+        String dataAccessRole = accountMapping.getUserMappings().get("hive");
+        // Surprisingly there is no ranger in the IDBroker mapping, so we can use solr
+        String rangerAuditRole = accountMapping.getUserMappings().get("solr");
+
+        return String.format("Services running on the cluster were unable to write to %s location. " +
+                        "This problem usually occurs due to cloud storage permission misconfiguration. " +
+                        "Services on the cluster are using Data Access Identity (%s) and Ranger Audit Identity ( %s) to write to the " +
+                        "Ranger Audit location (%s), therefore please verify that these roles have write access to this location. " +
+                        "During Data Lake cluster creation, CDP Control Plane attaches Data Access Assumer Identity (%s) to the IDBroker VM. " +
+                        "IDBroker will then use it to attach the other managed identities to the IDBroker VM, therefore Data Access Identity (%s) " +
+                        "permissions must, at a minimum, allow to attach the Data Access Identity and Ranger Access Identity. " +
+                        "Refer to Cloudera documentation at %s for the required rights.",
+                auditLocation, dataAccessRole, rangerAuditRole, auditLocation, managedIdentity, managedIdentity,
+                "https://docs.cloudera.com/management-console/cloud/environments-azure/topics/mc-az-minimal-setup-for-cloud-storage.html");
+    }
+
+    private String getRangerAuditDir(CloudStorage cloudStorage) {
+        if (cloudStorage.getLocations() != null) {
+            for (StorageLocation location : cloudStorage.getLocations()) {
+                if (CloudStorageCdpService.RANGER_AUDIT == location.getType()) {
+                    return location.getValue();
+                }
+            }
+        }
+        return "";
```

Comment:
Should not happen when launching via Datalake API (even if it did, it would be a programming bug), might be only broken when someone directly pokes StackV4 or ClusterV4. Perhaps made sense to at least `LOGGER.error()` this.

Reply:
Right, anything can be null or empty so, I am afraid, I need to reimplement the validation here :(. I am not sure that it is worth the effort, I just wanted to avoid NPE.

Reply:
Sounds reasonable. 👍